frosthaven_assistant/lib/Resource/action_handler.dart

```diff
@@ -62,10 +65,8 @@ class ActionHandler {
         //make sure to invalidate and rebuild all ui, since references will be broken
         getIt<GameState>().updateForUndo.value++;
       }
-
-
     } else {
-      getIt<Network>().client.send("undo");
+      _communication.sendToAll("undo");
     }
```

Comment:
Client should only send undo/redo to server, that then updates the state for all clients. looking closer you'd see the clients don't even handle undo or redo. Of course if you have some plan for change then that's another matter

Reply:
Good question. Sorry, I didn’t see your message for some reason…
The beauty of it is that client knows only about server, and server socket is all sockets that it is going to send to 😊

Reply:
aha. sorry, was confusing me. I'll merge now